cmd/crowdsec-cli/capi.go

```diff
@@ -180,12 +205,22 @@ func QueryCAPIStatus(hub *cwhub.Hub, credURL string, login string, password stri
 		Scenarios: scenarios,
 	}
 
-	_, _, err = client.Auth.AuthenticateWatcher(context.Background(), t)
+	authResp, _, err := client.Auth.AuthenticateWatcher(context.Background(), t)
 	if err != nil {
-		return err
+		return false, false, err
 	}
 
-	return nil
+	if err = client.GetClient().Transport.(*apiclient.JWTTransport).Expiration.UnmarshalText([]byte(authResp.Expire)); err != nil {
+		return true, false, err
+	}
+
+	client.GetClient().Transport.(*apiclient.JWTTransport).Token = authResp.Token
+
+	if client.IsEnrolled() {
+		return true, true, nil
+	} else {
```

Comment:
you can drop the else branch around the return, it will eventually be enforced by a linter, if it's not already, because it has led to unnecessary complexity